### DISCUSSION TITLE
动态表单

### DISCUSSION THREAD

**Discussion by yutao98712** [D-xuanmo/dl-ui#19] [Q&A]
请问，动态表单如何实现。后台接口返回json数据后，我如何重新渲染表单。目前似乎没有找到重新渲染的方法。

**Comment by D-xuanmo** [comment]
> 请问，动态表单如何实现。后台接口返回json数据后，我如何重新渲染表单。目前似乎没有找到重新渲染的方法。

可以参考这个 demo 中的 `formModel` 变量： https://uoo.ink/Form

<img width="1401" alt="QQ20240108-174025@2x" src="https://github.com/D-xuanmo/dl-ui/assets/20230104/a14a20dc-c538-4cfb-adee-4f4764582320">

**Reply by yutao98712** [reply]
不好意思，我可能没有讲清楚，我现在的问题是：我的formModel一开始是空的，然后formModel这个变量的值我需要通过后台接口来获取。如果formModel我前端写死，自然是没什么问题，表单时可以正常显示的，但是从接口获取后再赋值给formModel，表单无法显示出来，页面一片空白。所以我猜测是，表单挂载之后，再去改formModel是没用的，但我就是需要通过后台接口来动态生成表单项，这个有办法解决吗？

**Reply by yutao98712** [reply]
没事了，我已经解决了，通过componentKey和forceUpdate.

**Reply by D-xuanmo** [reply]
目前一个 model 对象，是所有属性进行了响应式转换的，具体是那个没有生效呢，可以给个简易 demo 看下，一般不建议使用强制刷新来做：https://run.xuanmo.xin/
